darwin/future/data_objects/advanced_filters.py

```diff
@@ -88,12 +88,16 @@ class NotContains(BaseMatcher):
 
 class SubjectFilter(BaseModel):
     subject: str
-    matcher: BaseMatcher
+    matcher: SerializeAsAny[BaseMatcher]
```

Comment:
This is an important change and should be noted for anyone else doing pydantic object inheritance. BaseMatcher is basically never meant to be used singularly, it acts like an ABC. When pydantic model_dump is used, it wonly only ever serialize fields that exist on BaseMatcher, so if you inherit from that and add fields, those added fields won't be serialized. The solution is to add in pydantics SerializeAsAny wrapper